tools/linker/CoreOptimizeGeneratedCode.cs

```diff
@@ -84,18 +105,42 @@ public override void Initialize (LinkContext context)
 					is_arm64_calling_convention = false;
 				}
 			}
+#if NET
+			this.context = context;
```

Comment:
Maybe add an `ExceptionalMarkHandler.Initialize (LinkContext)` method that stores the context? That would make the .NET-specific code smaller.

Reply:
This should also fix https://github.com/xamarin/xamarin-macios/pull/11374#issuecomment-829557041.